### PR TITLE
[uss_qualifier] netrid: net0260 check SP reports correct speed

### PR DESCRIPTION
**Pull request by Shastick**
Effort towards #754

### CODE REVIEW

**Review by mickmis — COMMENTED** [review]
Outside of minor comments and open question, LGTM.

**Review by mickmis — COMMENTED** [review]
Actually having second thoughts: we have already implementation of a check for validity of speed according to common dictionary: https://github.com/interuss/monitoring/blob/b804cf4e7cfe5970c0c5c0c2bb748aaa7c853985/monitoring/uss_qualifier/scenarios/astm/netrid/common_dictionary_evaluator.py#L427

Shouldn't we have a common implementation here? And just have separate .md docs to cover the different requirements depending on context (i.e. here NET0260 for SP).

File: monitoring/uss_qualifier/scenarios/astm/netrid/display_data_evaluator.py [hunk 1]
```diff
@@ -918,6 +921,25 @@ def _evaluate_normal_sp_observation(
                             details=f"{mapping.injected_flight.uss_participant_id}'s flight with injection ID {mapping.injected_flight.flight.injection_id} in test {mapping.injected_flight.test_id} had telemetry index {mapping.telemetry_index} at {injected_telemetry.timestamp} with lat={injected_telemetry.position.lat}, lng={injected_telemetry.position.lng}, alt={injected_telemetry.position.alt}, but Service Provider reported lat={observed_position.lat}, lng={observed_position.lng}, alt={observed_position.alt} at {mapping.observed_flight.query.query.request.initiated_at}",
                         )
 
+            if mapping.observed_flight.flight.raw.current_state is not None:
```
[the remark this answers is not in the document]
Comment (Shastick) [reply monitoring/uss_qualifier/scenarios/astm/netrid/display_data_evaluator.py:924]: `NET0450` is for display providers: the logic on this line is for a service provider in `_evaluate_normal_sp_observation` which is used to evaluate an observation made directly on a SP.

As to whether we should fail if current_state is missing, the OpenAPI specification, regarding the `current_state` field of an `RIDFlight` retrieved from a SP states that:

> The most up-to-date state of the aircraft.  Required when the aircraft is
            currently in the requested area unless no telemetry is expected during during
            the flight and `volumes` is specified.
> If current data is not being received from the UAS by the Service Provider,
            the lack of change in this field is sufficient to indicate that current
            data is not being received.
> If the USS believes a flight is airborne and expects to receive telemetry
            at some point during the flight but has not yet received telemetry, this
            field should be populated with the USS's best estimate of the position
            of the flight with appropriate accuracy indications.
            
I _guess_ that, given that we asked the USS to inject a flight, it should at least _believe_ that a flight is airborne, although, given that it is possible to inject flights that will only take place in the future, we may still expect to query the SP at a time where no flight has started yet (although, that is probably not the case here, as the particular logic we are looking at is comparing injected data versus observed one).

The existing check for the `alt` field will[ also not be run](https://github.com/interuss/monitoring/blob/613d1a693f62b77202b8bac17b9ff1a9f50011f4/monitoring/monitorlib/fetch/rid.py#L207) if the `current_state` is not provided.

We can probably revisit this separately, ideally once all the checks that are looking into the `current_state` field are merged? I'll mention this at the meeting tonight in any case.

[the remark this answers is not in the document]
Comment (mickmis) [reply monitoring/uss_qualifier/scenarios/astm/netrid/display_data_evaluator.py]: > NET0450 is for display providers: the logic on this line is for a service provider in _evaluate_normal_sp_observation which is used to evaluate an observation made directly on a SP.

Right!

> We can probably revisit this separately, ideally once all the checks that are looking into the current_state field are merged?

:+1:

**Comment by Shastick** [comment]
> Shouldn't we have a common implementation here? And just have separate .md docs to cover the different requirements depending on context (i.e. here NET0260 for SP).

After having had a look: at least `_evaluate_speed` seems to be specifically tailored for the display provider. (Ie, with v19 there is no requirement for the DP to report the speed, at least if I can trust the code – and the doc).

For the SP however, the requirement for the presence of the speed field are slightly different (and present for both v19 and v22a), so all speed-related requirements were not created equal.

I'm still in favor of getting all the requirements that involve looking into the state merged and then looking at where they will best belong.

**Review by mickmis — APPROVED** [review]
LGTM
Duplication issues discussed offline: TODO in code + issue to track coming back in bulk on it.

**Pull request merged**